lib/common/zstd_internal.h

```diff
@@ -238,8 +253,10 @@ void ZSTD_wildcopy(void* dst, const void* src, ptrdiff_t length, ZSTD_overlap_e
 	 * one COPY16() in the first call. Then, do two calls per loop since
 	 * at that point it is more likely to have a high trip count.
          */
+#ifndef __aarch64__
```

Comment:
With this change on a pixel-2 compiled with android NDK r20 I see a 5-15% regression in decompression speed.

Keeping this shortcut is important for two reasons:
1. Most copies are <= 16 bytes, so doing a 2nd copy is wasteful until they become long.
2. This shortcut separates out the branch statistics for the first iteration `(op >= oend)` is nearly always true, from the loop, which is more likely to have a high trip count.

Without this change, I see neutral compression and decompression speed on the Pixel 2. If you remove this change do you still get positive compression performance?

Reply:
> With this change on a pixel-2 compiled with android NDK r20 I see a 5-15% regression in decompression speed.
> 
> Keeping this shortcut is important for two reasons:
> 
> 1. Most copies are <= 16 bytes, so doing a 2nd copy is wasteful until they become long.
> 2. This shortcut separates out the branch statistics for the first iteration `(op >= oend)` is nearly always true, from the loop, which is more likely to have a high trip count.
> 
> Without this change, I see neutral compression and decompression speed on the Pixel 2. If you remove this change do you still get positive compression performance?

Without this change, there is still 1.5% positive compression on gcc9.2.0 , and almost on clang9.0.0 .

Reply:
We need to keep this code on aarch64 for the pixel2 benchmarks.

We could also try:

```c
#ifdef __aarch64__
do {
    COPY16(op, ip);
while (op < oend);
#else 
COPY16(op, ip);
if (op >= oend) return;
do {
    COPY16(op, ip);
    COPY16(op, ip);
while (op < oend);
#endif
```

Basically deleting the first if branch, but shortening the loop. I haven't measured it, so I don't know if it will work, but it is an idea.

Reply:
  | gcc-9.2.0 | clang-9.0.0
-- | -- | --
  | Compression | Decompression | Compression | Decompression
this modifcation | 1.11% | -3.58% | 0.05% | -1.05%
this   modifcationn +      neon function of ZSTD_copy8 and ZSTD_copy16 | 1.52% | -1.14% | -0.54% | -0.44%